### PR TITLE
jit_kernel/dsa: import cutlass CuTe-DSL runner on CUDA only

### PR DESCRIPTION
**Pull request by ckvermaAI**
The DSA CuTe-DSL runner (cutedsl_paged_mqa_logits) unconditionally imports cutlass, which is CUDA-only. It was imported eagerly on every non-ROCm platform, so on XPU the import aborted the deepseek_v2 -> deepseek_ocr import
chain, leaving DeepseekOCRForCausalLM unregistered and breaking model loading.

Gate the eager import positively on is_cuda() in both jit_kernel/dsa/__init__ and srt/layers/attention/dsa/dsa_indexer, so only CUDA imports it and all other platforms (ROCm, XPU, CPU) take the no-import path (pick_dsl_expand is set to None in dsa_indexer, its sole consumer). CUDA behavior is unchanged.

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :x: [Run #29078877829](https://github.com/sgl-project/sglang/actions/runs/29078877829)
Latest PR Test (Extra): :white_check_mark: [Run #29078877639](https://github.com/sgl-project/sglang/actions/runs/29078877639)

### CODE REVIEW

File: python/sglang/jit_kernel/dsa/__init__.py [hunk 1]
```diff
@@ -7,9 +7,20 @@
     deepgemm_paged_mqa_logits_split,
 )
 
+CuteDSLPagedMQALogitsRunner = None
+pick_dsl_expand = None
 if not is_hip():
```
Comment (polisettyvarma) [inline python/sglang/jit_kernel/dsa/__init__.py]: why not add xpu ?

**Review by polisettyvarma — CHANGES_REQUESTED** [review]

**Review by polisettyvarma — APPROVED** [review]

**Comment by ckvermaAI** [comment]
@mingfeima Please review this PR

**Comment by polisettyvarma** [comment]
due to https://github.com/sgl-project/sglang/pull/25220 and similar fix by AMD https://github.com/sgl-project/sglang/pull/30374

File: python/sglang/jit_kernel/dsa/__init__.py [hunk 2]
```diff
@@ -7,8 +7,13 @@
...
 
-if not is_hip():
-    # Preserve the original eager import behavior on non-ROCm platforms.
+CuteDSLPagedMQALogitsRunner = None
+pick_dsl_expand = None
+# The CuTe DSL runner requires `cutlass` (CUDA-only), so it is imported eagerly
+# only on platforms that provide it. ROCm and XPU leave the names as None
+# (mirrors the ROCm guard in srt/layers/attention/dsa/dsa_indexer.py) so
+# importing this package does not abort on those platforms.
+if not is_hip() and not is_xpu():
```
Comment (polisettyvarma) [inline python/sglang/jit_kernel/dsa/__init__.py]: seems cpu also requires this, instead import for CUDA only

**Review by polisettyvarma — CHANGES_REQUESTED** [review]

File: python/sglang/jit_kernel/dsa/__init__.py (hunk 2 above)
Comment (ZailiWang) [reply python/sglang/jit_kernel/dsa/__init__.py]: Hi @b8zhong  would you help confirm if this should only applied to CUDA?
If so, we can change the logic to `if is_cuda()` instead of a list of `if not is_pf1() and not is_pf2() ...`

Comment (ckvermaAI) [reply python/sglang/jit_kernel/dsa/__init__.py]: yes, we are changing to `if is_cuda()`

**Review by polisettyvarma — APPROVED** [review]

File: python/sglang/jit_kernel/dsa/__init__.py (hunk 2 above)
Comment (b8zhong) [reply python/sglang/jit_kernel/dsa/__init__.py]: Yes. Sorry about that.

**Review by mingfeima — APPROVED** [review]

**Comment by mingfeima** [comment]
@ZailiWang cpu ci still failed but seems a new fail: https://github.com/sgl-project/sglang/actions/runs/29014880447/job/86107367030?pr=30611

please take a look. we can also skip the deepseek v2 lite fp8 test for now and switch to others later - not a very good candidate for current test suite.

**Comment by b8zhong** [comment]
@mingfeima It's solved by merging in main

**Comment by mingfeima** [comment]
@ckvermaAI need rebase to fix the conflict.

**Comment by ZailiWang** [comment]
I'm checking with the `test_spec_eagle_topk_cpu.py` author why such error is raised in Xeon CI: https://github.com/sgl-project/sglang/actions/runs/29078877624/job/86316859248?pr=30611

**Comment by ZailiWang** [comment]
It's a test model downloading issue. The author has created the test case reverting PR #30772 . Would you please update with the changes in #30772 into this PR , then verify if Xeon CI can be restored? Many thanks.

**Comment by ckvermaAI** [comment]
Fix is merged in upstream: https://github.com/sgl-project/sglang/pull/30627 ; closing this PR
